### ISSUE TITLE
What is a block?

### ISSUE THREAD

**Issue by tbekolay** [PyCQA/pylint#2452]
## Question

Apologies if this has been asked before, but I'm curious what pylint considers a "block" and whether that could be documented more explicitly. Right now, the [docs](http://pylint.pycqa.org/en/latest/faq.html#is-it-possible-to-locally-disable-a-particular-message) state:

> add "#pylint: disable=some-message,another-one" at the desired block level or at the end of the desired line of code

I ask this because I was surprised to see that the following does not work:

```python
        if hasattr(SafeConfigParser, "read_file"):
            return SafeConfigParser.read_file(self, fp, filename)
        else:
            # pylint: disable=deprecated-method
            return SafeConfigParser.readfp(self, fp, filename)
```

The `deprecated-method` warning is still raised.

However, the following _does_ work:

```python
        try:
            return SafeConfigParser.read_file(self, fp, filename)
        except AttributeError:
            # pylint: disable=deprecated-method
            return SafeConfigParser.readfp(self, fp, filename)
```

The `deprecated-method` warning is disabled, as desired. In the `if/else` case, I had to add the `# 
 pylint: disable=` line to block that contains the `if/else` instead.

**Comment by Pierre-Sassoulas** [comment]
The bug is a duplicate of #872, the documentation need to be done though.

```python
        if hasattr(SafeConfigParser, "read_file"):
            return SafeConfigParser.read_file(self, fp, filename)
        else:  # pylint: disable=deprecated-method
            return SafeConfigParser.readfp(self, fp, filename)
```
Would work here.
